### PR TITLE
optimize:  optimize exceptionHandler code logic

### PR DESCRIPTION
**Pull request by YuKongEr**
### Ⅰ. Describe what this PR did
Optimizi code logic

### CODE REVIEW

**Comment by objcoding** [comment]
This part of the code has been refactored in https://github.com/seata/seata/pull/2486

**Comment by YuKongEr** [comment]
ok i think  `AbstractTransactionRequest#exceptionHandleTemplate()` use generic method  would be better

**Comment by objcoding** [comment]
> ok i think `AbstractTransactionRequest#exceptionHandleTemplate()` use generic method would be better

What you said should be AbstractExceptionHandler#exceptionHandleTemplate.

File: core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java [hunk 1]
```diff
@@ -111,8 +111,7 @@ public void onException(T request, S response, Exception rex) {
      * @param request  the request
      * @param response the response
      */
-    public void exceptionHandleTemplate(Callback callback, AbstractTransactionRequest request,
-        AbstractTransactionResponse response) {
+    public <T extends AbstractTransactionRequest, S extends AbstractTransactionResponse> void exceptionHandleTemplate(Callback<T, S> callback, T request, S response) {
```
Comment (objcoding) [inline core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:114]: I think this optimization is ok.

Comment (YuKongEr) [reply core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:114]: ok i will change commit

Comment (YuKongEr) [reply core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:114]: done

Comment (objcoding) [reply core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:114]: Please click the resolve button.

**Comment by slievrly** [comment]
@YuKongEr Thank you for your contribution.

**Pull request merged**